src/main/java/xyz/holocons/mc/holoitemsrevamp/item/MagnetBook.java

```diff
@@ -65,7 +58,7 @@ public Enchantment getEnchantment() {
     @Override
     public ItemStack applyEnchantment(ItemStack itemStack) {
         var enchantedStack = itemStack.clone();
-        var enchantedMeta = enchantedStack.hasItemMeta() ? enchantedStack.getItemMeta() : Bukkit.getItemFactory().getItemMeta(enchantedStack.getType());
+        var enchantedMeta = (EnchantmentStorageMeta) enchantedStack.getItemMeta();
 
         if (enchantedMeta.addEnchant(getEnchantment(), 1, false)) {
```

Comment:
Should be `EnchantmentStorageMeta#addStoredEnchant` instead of `ItemMeta#addEnchant`